File: app/src/main/java/suzp1984/github/io/exapidemo/app/fragments/ListViewSample1Activity.java
package suzp1984.github.io.exapidemo.app.fragments;

import android.view.View;

/**
 * Created by suzhenxi on 10/19/2016.
 */

public class ListViewSample1Activity extends FragmentInListViewActivity {

    @Override
    protected void setupListView() {
        mAdd.setVisibility(View.GONE);
        mRemove.setVisibility(View.GONE);

        ListAdapter1 adapter = new ListAdapter1(getSupportFragmentManager());

        mListView.setAdapter(adapter);

        adapter.registerFragmentAdapterDataObserver(new BaseListFragmentAdapter.FragmentAdapterDataObserver() {
            @Override
            public void onViewCountChanged(int count) {
                super.onViewCountChanged(count);

                mViewHolderCount.setText(String.valueOf(count));
            }

            @Override
            public void onAttachedFragmentCountChanged(int count) {
                super.onAttachedFragmentCountChanged(count);

                mFragmentsCount.setText(String.valueOf(count));
            }
        });
    }
}

File: app/src/main/kotlin/suzp1984/github/io/exapidemo/app/fragments/ListViewSample1Activity.kt
package suzp1984.github.io.exapidemo.app.fragments

import android.view.View

/**
 * Created by moses on 10/19/2016.
 */

class ListViewSample1Activity : FragmentInListViewActivity() {

    override fun setupListView() {
        mAdd.visibility = View.GONE
        mRemove.visibility = View.GONE

        val adapter = ListAdapter1(supportFragmentManager)

        mListView.adapter = adapter

        adapter.registerFragmentAdapterDataObserver(object : BaseListFragmentAdapter.FragmentAdapterDataObserver() {
            override fun onViewCountChanged(count: Int) {
                super.onViewCountChanged(count)

                mViewHolderCount.text = count.toString()
            }

            override fun onAttachedFragmentCountChanged(count: Int) {
                super.onAttachedFragmentCountChanged(count)

                mFragmentsCount.text = count.toString()
            }
        })
    }
}
